chain/src/main/java/org/ethereum/beacon/chain/MutableBeaconChain.java

```diff
@@ -4,11 +4,21 @@
 
 public interface MutableBeaconChain extends BeaconChain {
 
+  enum ImportResult {
+    OK,
+    ExistingBlock,
+    NoParent,
+    ExpiredBlock,
+    InvalidBlock,
+    StateMismatch,
+    OtherError
```

Comment:
I'd throw out `OtherError` until it is needed. Maybe we could keep staying with more concrete results.

Reply:
IMO, `UnexpectedError` reads better.

Reply:
And I think we definitely need a `StateTransitionError` to reflect `SpecAssertionException` that is thrown by the spec.